File: src/main/java/arcana/common/research/requirements/XpRequirement.java
package arcana.common.research.requirements;

import arcana.common.research.Requirement;
import net.minecraft.entity.player.PlayerEntity;

public class XpRequirement implements Requirement {
    int minLevel, takeLevel;
    @Override
    public boolean satisfied(PlayerEntity player) {
        return player.experienceLevel >= minLevel;
    }

    @Override
    public void take(PlayerEntity player) {
        player.giveExperienceLevels(-takeLevel);
    }

    @Override
    public void deserialize(String data) {
        String[] parts = data.split("-");
        minLevel = Integer.parseInt(parts[0]);
        takeLevel = Integer.parseInt(parts[1]);
    }
}

File: src/main/kotlin/arcana/common/research/requirements/XpRequirement.kt
package arcana.common.research.requirements

import arcana.common.research.Requirement
import net.minecraft.entity.player.PlayerEntity

class XpRequirement : Requirement {
    var minLevel = 0
    var takeLevel = 0
    override fun satisfied(player: PlayerEntity): Boolean {
        return player.experienceLevel >= minLevel
    }

    override fun take(player: PlayerEntity) {
        player.giveExperienceLevels(-takeLevel)
    }

    override fun deserialize(data: String) {
        val parts = data.split("-")
        minLevel = parts[0].toInt()
        takeLevel = parts[1].toInt()
    }
}